### PR TITLE
Dropblock

### PR DESCRIPTION
**Pull request by dpodvyaznikov**
Add dropout layer for convolutions based on this [paper](https://arxiv.org/pdf/1810.12890.pdf)

### CODE REVIEW

File: batchflow/models/tf/layers/drop_block.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    """
+    """
+    shape = inputs.shape.as_list()
+    n_dims = len(shape)
+
+    # Transpose
+    if data_format == 'channels_first':
+        inputs = tf.transpose(inputs, perm=[0, *list(range(2, n_dims)), 1])
+
+    spatial_dims, channels = shape[1:-1], shape[-1]
```
Comment (roman-kh) [inline batchflow/models/tf/layers/drop_block.py]: `spatial_dims` can be `(None, None)`

File: batchflow/models/tf/layers/drop_block.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import numpy as np
+import tensorflow as tf
+
+# TODO:
+# Add scheduling scheme to gradually decrease keep_prob from 1 to target value.
+
+def _dropblock(inputs, keep_prob, block_size, seed, data_format):
+    """
+    """
+    shape = inputs.shape.as_list()
```
Comment (roman-kh) [inline batchflow/models/tf/layers/drop_block.py]: use `tf.shape()`

File: batchflow/models/tf/layers/drop_block.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    n_dims = len(shape)
+
+    # Transpose
+    if data_format == 'channels_first':
+        inputs = tf.transpose(inputs, perm=[0, *list(range(2, n_dims)), 1])
```
[the remark this answers is not in the document]
Comment (dpodvyaznikov) [reply batchflow/models/tf/layers/drop_block.py]: Affects more than I expected, moved to explicit NCHW handling.

File: batchflow/models/tf/layers/drop_block.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+# Add scheduling scheme to gradually decrease keep_prob from 1 to target value.
+
+def dropblock(inputs, dropout_rate, block_size, is_training, data_format, seed=None):
+    """ Drop Block module.
+
+    Parameters
+    ----------
+    inputs : tf.Tensor
+        Input tensor
+    dropout_rate : float
```
[the remark this answers is not in the document]
Comment (dpodvyaznikov) [reply batchflow/models/tf/layers/drop_block.py]: It can indeed. Now it can also be callable.

**Pull request merged**